builtin-functions/_functions.txt

```diff
@@ -702,6 +693,7 @@ function substr_compare ($main_str ::: string, $str ::: string, $offset ::: int,
 function str_starts_with ($haystack ::: string, $needle ::: string) ::: bool;
 function str_ends_with ($haystack ::: string, $needle ::: string) ::: bool;
 
+/** @kphp-internal-param-readonly $s */
 function trim ($s ::: string, $what ::: string = " \n\r\t\v\0") ::: string;
 function ltrim ($s ::: string, $what ::: string = " \n\r\t\v\0") ::: string;
```

Comment:
ltrim, rtrim — also readonly?

Reply:
That would require a special overloading for both ltrim and rtrim.
I think it can be done in a separate PR, since it's a routine task that doesn't bring anything new to the table except it requires more testing and code to be reviewed in this (quite risky) PR.